packages/block/src/header.ts

```diff
@@ -325,30 +377,31 @@ export class BlockHeader {
    * Returns the hash of the block header.
    */
   hash(): Buffer {
-    return rlphash(this.raw)
+    const values: Buffer[] = [
+      this.parentHash,
+      this.uncleHash,
+      this.coinbase,
+      this.stateRoot,
+      this.transactionsTrie,
+      this.receiptTrie,
+      this.bloom,
+      this.difficulty,
+      unpadBuffer(this.number),
```

Comment:
Is only the number unpadded?

Reply:
I think we should unpad it directly in the constructor, so `isGenesis` is also correct in case we use `Buffer.from('00', 'hex')` as number.

Reply:
Per yellow paper the following vars should be treated as number and should be unpadded; `gasLimit`, `difficulty`, `gasUsed`, `number`, `timestamp`

![image](https://user-images.githubusercontent.com/29359032/94161034-8db86980-fe85-11ea-9937-98e9ef3c544b.png)

Reply:
This is now solved in the constructor of `Header`, dd228bc

Reply:
Relevant Yellow Paper reference regarding stripping zeros:

![image](https://user-images.githubusercontent.com/29359032/94175281-c3b21980-fe96-11ea-8ede-ad269e5b8a67.png)

Reply:
👍  Going along with everything here, this number unpadding was just a quick start to get the hash correct.